### PR TITLE
ERT / nukies can now choose any NON KARMA species

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Adds a new magic mirror type that can only select non karma species, and only lets body be adjusted
**NOTE, DUE TO HOW THE APPEARANCE CHANGER WORKS, ADMINS WILL ALWAYS BE ABLE TO SELECT KARMA SPECIES**, so de-admin first to test, and hope that admins don't pick vox / slime on the live server.

ERT now can pick their non karma species after gender. If species is not picked, defaults to human.
Slight more ERT hair colour options, blue is now a proper eye colour to be picked

ERT signup time down to 45 seconds from 60, since we have 10 seconds added from species picking, we don't want to make it too long, and no one needs 1 minute to decide whether to sign up for a role.

## Why It's Good For The Game

Now that species are more even damage wise, by chosing non karma species only, we can open up the floodgates for non karma species for ERT / nukies.

This gets us more variation in our teams, considering we do not have job restrictions for command (or CC) like other servers. We have the sprites for it, we have the higher up roles not being pure human, so we should do it.
Also means nukies can go undercover on station and not get called out for being 5 human vips trying to visit command.

We will not have people snowflaking their ERT memebers to be their own characters, as they do not get to name themself, nor have a magic mirror to spend 5 minutes of customization on as station needs help,

My one concern is diona, due to healing. If balance team / public wants it, it can be removed from the list of options.

## TODO

- [x] Check if elite hardsuit is broken for skrell, cry if it is
- [x] See how people feel about diona
- [x] cry as elite hardsuit IS broken for skrell
- [x] get someone to sprite it or sprite it yourself

##  New Elite sprite for skrell (was none before, antenna is slightly darker on side view in current version)
https://media.discordapp.net/attachments/586307829087076362/1057868484303913001/image.png

## Testing

Ensure mirror worked de-admined.
Spawned as each species type through ert panel, to ensure it worked right / looked right / did not runtime.

## Changelog
:cl:
add: Added a magic mirror in the syndicate nuclear base that lets them choose a NON KARMA SPECIES / edit their body, without being able to change name or voice.
add: ERT can now choose a NON KARMA SPECIES before spawning in to be. They do not have a magic mirror.
tweak: ERT sign up time is now 45 seconds instead of 60 seconds.
/:cl:

### CODE REVIEW

**Comment by Komrad822** [comment]
BAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAASED
AS FUUUUUUUUUUUUUUUUUUUUCK PR.

**Comment by TravisAngeI** [comment]
im interested to see how this works out, could bring a lot more color to ERTs and nukies

**Comment by Eric6426-Blossom** [comment]
This PR will spawn mothnukies turning off the gravgen 👍 Cursed

Jokes aside, is there any chance you could add some kinda warning or something to picking karma races for admins
Since I doubt for ERT but Nukies I fear someone might do the oopsie of picking a karma race.

**Comment by Coolrune206** [comment]
Will this result in species-specific ERT names? Or will the Diona ERT be called Sergeant Bob (not that that isn't inherently funny).

**Comment by Qwertytoforty** [comment]
> Will this result in species-specific ERT names? Or will the Diona ERT be called Sergeant Bob (not that that isn't inherently funny).

Human names are used, think of them as just team names I guess. We don't have last names for all the non karma species, so it would end up looking strange.

**Comment by SpringSkipper** [comment]
> This PR will spawn mothnukies turning off the gravgen 👍 Cursed
> 
> Jokes aside, is there any chance you could add some kinda warning or something to picking karma races for admins Since I doubt for ERT but Nukies I fear someone might do the oopsie of picking a karma race.

Get those flashes and flyswatters ready...

**Comment by Komrad822** [comment]
This may be out of scope, but would it be a good idea to add mirror for SIT and SST areas aswell?

Also, I know dsquad is meant to end the round as quickly as possible adding another option delays them a bit, but would the race option be included for them too?

File: code/game/objects/structures/mirror.dm [hunk 1]
```diff
@@ -167,3 +173,11 @@
 
 /obj/structure/mirror/magic/proc/curse(mob/living/user)
 	return
+
+/obj/structure/mirror/magic/nuclear
+	name = "M.A.G.I.C mirror"
+	desc = "The M.A.G.I.C mirror will let you change your species in a flash! Be careful, any implants (not biochips) in you will be destroyed on use."
```
Comment (Sirryan2002) [inline code/game/objects/structures/mirror.dm:179]: this part worries me, Gamma/Red ERTs will end up destroying their nutriment pumps or combat implants a lot b/c of this

File: code/modules/response_team/ert.dm [hunk 2]
```diff
@@ -89,17 +89,25 @@ GLOBAL_VAR_INIT(ert_request_answered, FALSE)
 	var/list/ert_gender_prefs = list()
 	for(var/mob/M in GLOB.response_team_members)
 		ert_gender_prefs.Add(input_async(M, "Please select a gender (10 seconds):", list("Male", "Female")))
-	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(get_ert_role_prefs), GLOB.response_team_members, ert_gender_prefs), 100)
+	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(get_ert_species_prefs), GLOB.response_team_members, ert_gender_prefs), 100)
```
Comment (Sirryan2002) [inline code/modules/response_team/ert.dm:92]:
```suggestion
	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(get_ert_species_prefs), GLOB.response_team_members, ert_gender_prefs), 10 SECONDS)
```

File: code/modules/response_team/ert.dm [hunk 3]
```diff
@@ -89,17 +89,25 @@ GLOBAL_VAR_INIT(ert_request_answered, FALSE)
...
 
-/proc/get_ert_role_prefs(list/response_team_members, list/ert_gender_prefs) // Why the FUCK is this variable the EXACT SAME as the global one
-	var/list/ert_role_prefs = list()
+/proc/get_ert_species_prefs(list/response_team_members, list/ert_gender_prefs)
 	for(var/datum/async_input/A in ert_gender_prefs)
 		A.close()
+	var/list/ert_species_prefs = list()
+	for(var/mob/M in GLOB.response_team_members)
+		ert_species_prefs.Add(input_async(M, "Please select a species (10 seconds):", list("Human", "Tajaran", "Skrell", "Unathi", "Diona", "Vulpkanin", "Nian", "Random")))
+	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(get_ert_role_prefs), GLOB.response_team_members, ert_gender_prefs, ert_species_prefs), 100)
```
Comment (Sirryan2002) [inline code/modules/response_team/ert.dm:100]:
```suggestion
	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(get_ert_role_prefs), GLOB.response_team_members, ert_gender_prefs, ert_species_prefs), 10 SECONDS)
```

File: code/modules/response_team/ert.dm [hunk 4]
```diff
@@ -89,17 +89,25 @@ GLOBAL_VAR_INIT(ert_request_answered, FALSE)
...
+
+/proc/get_ert_role_prefs(list/response_team_members, list/ert_gender_prefs, list/ert_species_prefs) // Why the FUCK is this variable the EXACT SAME as the global one
```
Comment (Sirryan2002) [inline code/modules/response_team/ert.dm:102]: why this is globally scoped is beyond me

File: code/game/objects/structures/mirror.dm (hunk 1 above)
Comment (Qwertytoforty) [reply code/game/objects/structures/mirror.dm:179]: Only nukies have the mirror!

It warns them on examine and on use

Comment (Sirryan2002) [reply code/game/objects/structures/mirror.dm:179]: ohhh, that makes a lot more sense

File: code/modules/response_team/ert.dm [hunk 5]
```diff
@@ -89,17 +89,25 @@ GLOBAL_VAR_INIT(ert_request_answered, FALSE)
...
+	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(get_ert_role_prefs), GLOB.response_team_members, ert_gender_prefs, ert_species_prefs), 10 SECONDS)
+
+/proc/get_ert_role_prefs(list/response_team_members, list/ert_gender_prefs, list/ert_species_prefs) // Why the FUCK is this variable the EXACT SAME as the global one
+	var/list/ert_role_prefs = list()
+	for(var/datum/async_input/A in ert_species_prefs)
+		A.close()
 	for(var/mob/M in response_team_members)
 		ert_role_prefs.Add(input_ranked_async(M, "Please order ERT roles from most to least preferred (20 seconds):", GLOB.active_team.get_slot_list()))
-	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(dispatch_response_team), response_team_members, ert_gender_prefs, ert_role_prefs), 200)
+	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(dispatch_response_team), response_team_members, ert_gender_prefs, ert_species_prefs, ert_role_prefs), 200)
```
Comment (Sirryan2002) [inline code/modules/response_team/ert.dm:108]:
```suggestion
	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(dispatch_response_team), response_team_members, ert_gender_prefs, ert_species_prefs, ert_role_prefs), 20 SECONDS)
```

File: code/modules/response_team/ert.dm [hunk 6]
```diff
@@ -134,35 +143,55 @@ GLOBAL_VAR_INIT(ert_request_answered, FALSE)
...
+	var/species = S.type
+	M.set_species(species, TRUE)
 	M.dna.ready_dna(M)
 	M.cleanSE() //No fat/blind/colourblind/epileptic/whatever ERT.
 	M.overeatduration = 0
-
-	var/hair_c = pick("#8B4513","#000000","#FF4500","#FFD700") // Brown, black, red, blonde
-	var/eye_c = pick("#000000","#8B4513","1E90FF") // Black, brown, blue
+	var/obj/item/organ/external/head/head_organ = M.get_organ("head")
+	var/eye_c = pick("#000000","#8B4513","#1E90FF") // Black, brown, blue
```
Comment (Sirryan2002) [inline code/modules/response_team/ert.dm:172]:
```suggestion
	var/eye_c = pick("#000000", "#8B4513", "#1E90FF") // Black, brown, blue
```

File: code/modules/response_team/ert.dm [hunk 7]
```diff
@@ -134,35 +143,55 @@ GLOBAL_VAR_INIT(ert_request_answered, FALSE)
...
 	var/skin_tone = rand(-120, 20) // A range of skin colors
 
-	head_organ.facial_colour = hair_c
-	head_organ.sec_facial_colour = hair_c
-	head_organ.hair_colour = hair_c
-	head_organ.sec_hair_colour = hair_c
+	switch(new_species) //Diona not included as they don't use the hair colours
+		if("Human", "Tajaran", "Vulpkanin", "Nian")
+			var/hair_c_htvn = pick("#8B4513","#000000","#FF4500","#FFD700", "#d4d1bf") // Brown, black, red, blonde, grey
```
Comment (Sirryan2002) [inline code/modules/response_team/ert.dm:177]:
```suggestion
			var/hair_c_htvn = pick("#8B4513", "#000000", "#FF4500", "#FFD700", "#d4d1bf") // Brown, black, red, blonde, grey
```

File: code/modules/response_team/ert.dm [hunk 8]
```diff
@@ -134,35 +143,55 @@ GLOBAL_VAR_INIT(ert_request_answered, FALSE)
...
+			head_organ.facial_colour = hair_c_htvn
+			head_organ.sec_facial_colour = hair_c_htvn
+			head_organ.hair_colour = hair_c_htvn
+			head_organ.sec_hair_colour = hair_c_htvn
+			M.skin_colour = hair_c_htvn
+		if("Skrell", "Unathi") //Some common skrell / unathi colours
+			var/list/su = list("#1f138b","#272525","#07a035","#8c00ff", "#a80c0c")
```
Comment (Sirryan2002) [inline code/modules/response_team/ert.dm:184]:
```suggestion
			var/list/su = list("#1f138b", "#272525", "#07a035", "#8c00ff", "#a80c0c")
```

File: code/modules/response_team/ert.dm [hunk 9]
```diff
@@ -134,35 +143,55 @@ GLOBAL_VAR_INIT(ert_request_answered, FALSE)
...
 		if(new_gender == "Male")
 			M.change_gender(MALE)
 		else
 			M.change_gender(FEMALE)
 
-	M.set_species(/datum/species/human, TRUE)
+	if(!new_species)
+		new_species = "Human"
+	if(new_species == "Random")
+		new_species = pick("Human", "Tajaran", "Skrell", "Unathi", "Diona", "Vulpkanin", "Nian", "Random")
```
Comment (S34NW) [inline code/modules/response_team/ert.dm:164]:
```suggestion
		new_species = pick("Human", "Tajaran", "Skrell", "Unathi", "Diona", "Vulpkanin", "Nian")
```

Comment (S34NW) [reply code/modules/response_team/ert.dm:164]: recursive random

**Review by S34NW — CHANGES_REQUESTED** [review]

**Review by Sirryan2002 — APPROVED** [review]

**Pull request merged**